### ISSUE TITLE
cups-pdf problem with Mac Yosemite 10.10.3 and permissions

### ISSUE THREAD

**Issue by jkleckner** [Homebrew/homebrew#40083]
Has anyone gotten cups-pdf to work with Mac OS 10.10.3 (Yosemite)?
I was seeing an error message:
```
kernel[0]: Sandbox: cups-pdf(6439) deny file-read-metadata /usr/local/etc/cups/cups-pdf.conf
```
Based on [this MacPorts issue](https://trac.macports.org/ticket/45006) and [this link](http://support.en.kodak.com/app/answers/detail/a_id/49247/selected/true), I added `Sandboxing relaxed` at the end of `/etc/cups/cups-files.conf` and changed the ```Out``` declaration in the file `/usr/local/etc/cups/cups-pdf.conf` to be `Out /var/spool/cups-pdf/${USER}` but this also didn't work, emitting the message: ```[ERROR] failed to set file mode for PDF file (non fatal) (/var/spool/cups-pdf/jim/job_75-homebrew__a__.pdf)``` (See the full debug log at the end of this).

[This person](https://bitbucket.org/codepoet/cups-pdf-for-mac-os-x/issue/55/file-doesnt-print#comment-17921875) got it to work by forcing the user to be root, which seems a bit of a hack and a security risk.

Before I sleuth any further, I thought I would ask others.

For the curious, the motivation for getting this to work on a mac, given that you can save as pdf from the print dialog, is to remove a password from a PDF file sent by my bank prior to filing into my encrypted store by printing it into cups-pdf and thereby capturing a decent non-bitmap-based copy.

### Debug log from cups-pdf:

I'm using ```brew``` version 0.9.5. and ran ```brew update``` and ```brew doctor``` before all of this.

> Mon May 25 17:57:47 2015  [DEBUG] switching to new gid (_lp)
> Mon May 25 17:57:47 2015  [DEBUG] initialization finished (v2.6.1)
> Mon May 25 17:57:47 2015  [DEBUG] user identified (jim)
> Mon May 25 17:57:47 2015  [DEBUG] output directory name generated (/var/spool/cups-pdf/jim)
> Mon May 25 17:57:47 2015  [DEBUG] user information prepared
> Mon May 25 17:57:47 2015  [DEBUG] spoolfile name created (/var/spool/cups-pdf/SPOOL/cups2pdf-3389)
> Mon May 25 17:57:47 2015  [DEBUG] source stream ready
> Mon May 25 17:57:47 2015  [DEBUG] destination stream ready (/var/spool/cups-pdf/SPOOL/cups2pdf-3389)
> Mon May 25 17:57:47 2015  [DEBUG] owner set for spoolfile (/var/spool/cups-pdf/SPOOL/cups2pdf-3389)
> Mon May 25 17:57:47 2015  [DEBUG] using traditional fgets
> Mon May 25 17:57:47 2015  [DEBUG] found beginning of postscript code (%!PS-Adobe-3.0)
> Mon May 25 17:57:47 2015  [DEBUG] now extracting postscript code
> Mon May 25 17:57:47 2015  [DEBUG] found title in ps code ((fix cups-pdf printing \302\267 ...516 \302\267 Homebrew/homebrew)
> aÿ)
> Mon May 25 17:57:47 2015  [DEBUG] found embedded (e)ps code (%!PS-TrueTypeFont-1.0000-1.0000-2)
> Mon May 25 17:57:47 2015  [DEBUG] found embedded (e)ps code (%!PS-TrueTypeFont-1.0000-1.0000-2)
> Mon May 25 17:57:47 2015  [DEBUG] found embedded (e)ps code (%!PS-TrueTypeFont-1.0000-0.0000-2)
> Mon May 25 17:57:47 2015  [DEBUG] found embedded (e)ps code (%!PS-TrueTypeFont-1.0000-1.0290-2)
> Mon May 25 17:57:47 2015  [DEBUG] found embedded (e)ps code (%!PS-TrueTypeFont-1.0000-5.2200-2)
> Mon May 25 17:57:47 2015  [DEBUG] found embedded (e)ps code (%!PS-TrueTypeFont-1.0000-1.0000-2)
> Mon May 25 17:57:47 2015  [DEBUG] found embedded (e)ps code (%!PS-TrueTypeFont-1.0000-1.0000-2)
> Mon May 25 17:57:47 2015  [DEBUG] found embedded (e)ps code (%!PS-TrueTypeFont-1.0000-0.0000-2)
> Mon May 25 17:57:47 2015  [DEBUG] found embedded (e)ps code (%!PS-TrueTypeFont-1.0000-1.0000-2)
> Mon May 25 17:57:47 2015  [DEBUG] found end of embedded (e)ps code (%%EOF)
> Mon May 25 17:57:47 2015  [DEBUG] all data written to spoolfile (/var/spool/cups-pdf/SPOOL/cups2pdf-3389)
> Mon May 25 17:57:47 2015  [DEBUG] trying to use PS title ((fix cups-pdf printing \302\267 ...516 \302\267 Homebrew/homebrew)
> aÿ)
> Mon May 25 17:57:47 2015  [DEBUG] removing trailing newlines from title ((fix cups-pdf printing \302\267 ...516 \302\267 Homebrew/homebrew)
> aÿ)
> Mon May 25 17:57:47 2015  [DEBUG] removing slashes from full title ((fix cups-pdf printing \302\267 ...516 \302\267 Homebrew/homebrew)
> aÿ)
> Mon May 25 17:57:47 2015  [DEBUG] removing special characters from title (homebrew)
> aÿ)
> Mon May 25 17:57:47 2015  [DEBUG] title successfully retrieved (job_75-homebrew__a__)
> Mon May 25 17:57:47 2015  [DEBUG] input data read from stdin
> Mon May 25 17:57:47 2015  [DEBUG] output filename created (/var/spool/cups-pdf/jim/job_75-homebrew__a__.pdf)
> Mon May 25 17:57:47 2015  [DEBUG] ghostscript commandline built (/usr/bin/pstopdf  -o /var/spool/cups-pdf/jim/job_75-homebrew__a__.pdf /var/spool/cups-pdf/SPOOL/cups2pdf-3389)
> Mon May 25 17:57:47 2015  [DEBUG] output file unlinked (/var/spool/cups-pdf/jim/job_75-homebrew__a__.pdf)
> Mon May 25 17:57:47 2015  [DEBUG] TMPDIR set for GhostScript (/tmp)
> Mon May 25 17:57:47 2015  [DEBUG] waiting for child to exit
> Mon May 25 17:57:47 2015  [DEBUG] entering child process
> Mon May 25 17:57:47 2015  [DEBUG] GID set for current user
> Mon May 25 17:57:47 2015  [DEBUG] supplementary groups set for current user
> Mon May 25 17:57:47 2015  [DEBUG] UID set for current user (jim)
> Mon May 25 17:57:47 2015  [DEBUG] ghostscript has finished (3328)
> Mon May 25 17:57:47 2015  [ERROR] failed to set file mode for PDF file (non fatal) (/var/spool/cups-pdf/jim/job_75-homebrew__a__.pdf)
> Mon May 25 17:57:47 2015  [DEBUG] ERRNO: 2
> Mon May 25 17:57:47 2015  [DEBUG] no postprocessing
> Mon May 25 17:57:47 2015  [DEBUG] spoolfile unlinked (/var/spool/cups-pdf/SPOOL/cups2pdf-3389)
> Mon May 25 17:57:47 2015  [DEBUG] all memory has been freed
> Mon May 25 17:57:47 2015  [STATUS] PDF creation successfully finished (jim)

**Comment by xu-cheng** [comment]
You should report this issue to upstream as it's not homebrew related.

**Comment by jkleckner** [comment]
I agree that it would be preferable for the upstream to fix it.

I did create a [patch](https://github.com/jkleckner/homebrew/commit/77de7dbb91e8e53e26831cfc4ff8bbb72fa0c101) based on the discussion here: https://bitbucket.org/codepoet/cups-pdf-for-mac-os-x/issue/55/file-doesnt-print#comment-17921875

It did work to create files but permissions still should be set to the printing user, not root.

It is here just in case someone else wants to carry this further for some reason.

The original use case I had to print a document from my bank was solved with ghostscript:
  http://www.commandlinefu.com/commands/view/4345/remove-security-limitations-from-pdf-documents-using-ghostscript
